java.lang.NoClassDefFoundError: org/json/JSONException on startup of 3.4-beta2

Extensions which have a dependency on org.json and depend on OpenRefine to include it rather than bundling it themselves will fail to initialize with the above NoClassDefFoundError. This is a general problem with dynamically loading extensions that have missing dependencies or other class loader issues.

Butterfly attempts to deal with module loading issues by catching `Exception` as a catch-all, but because this is an `Error`, it is not a subclass of `Exception`.

### Steps to reproduce the behavior:
1. Download 3.4-beta2 Windows kit
2. Unzip
3. Double click openrefine.exe
4. See error 

I killed it quickly because I realized that I wasn't sure I'd backed up my workspace directory, so not sure if it would have finished starting up.

### Current results

15:26:29.926 [            refine_server] Starting Server bound to '127.0.0.1:3333' (0ms)
15:26:29.984 [            refine_server] Initializing context: '/' from 'C:\Users\tfmorris\Downloads\openrefine-win-with-java-3.4-beta2\openrefine-3.4-beta2\webapp' (58ms)
...
Exception in thread "main" java.lang.NoClassDefFoundError: org/json/JSONException
        at java.lang.Class.getDeclaredMethods0(Native Method)
        at java.lang.Class.privateGetDeclaredMethods(Unknown Source)
        at java.lang.Class.privateGetPublicMethods(Unknown Source)
        at java.lang.Class.getMethods(Unknown Source)
        at org.mozilla.javascript.JavaMembers.discoverAccessibleMethods(JavaMembers.java:380)
        at org.mozilla.javascript.JavaMembers.discoverAccessibleMethods(JavaMembers.java:335)
        at org.mozilla.javascript.JavaMembers.reflect(JavaMembers.java:450)
        at org.mozilla.javascript.JavaMembers.<init>(JavaMembers.java:76)
        at org.mozilla.javascript.JavaMembers.lookupClass(JavaMembers.java:838)
        at org.mozilla.javascript.NativeJavaClass.initMembers(NativeJavaClass.java:84)
        at org.mozilla.javascript.NativeJavaClass.<init>(NativeJavaClass.java:78)
        at org.mozilla.javascript.NativeJavaPackage.getPkgProperty(NativeJavaPackage.java:164)
        at org.mozilla.javascript.NativeJavaPackage.get(NativeJavaPackage.java:114)
        at org.mozilla.javascript.ScriptableObject.getProperty(ScriptableObject.java:1617)
        at org.mozilla.javascript.ScriptRuntime.getObjectProp(ScriptRuntime.java:1437)
        at org.mozilla.javascript.ScriptRuntime.getObjectProp(ScriptRuntime.java:1423)
        at org.mozilla.javascript.gen.c10._c0(file:/C:/Users/tfmorris/AppData/Roaming/OpenRefine/extensions/rdf-extension/MOD-INF/controller.js:4)
        at org.mozilla.javascript.gen.c10.call(file:/C:/Users/tfmorris/AppData/Roaming/OpenRefine/extensions/rdf-extension/MOD-INF/controller.js)
        at org.mozilla.javascript.ContextFactory.doTopCall(ContextFactory.java:398)
        at org.mozilla.javascript.ScriptRuntime.doTopCall(ScriptRuntime.java:3065)
        at org.mozilla.javascript.gen.c10.call(file:/C:/Users/tfmorris/AppData/Roaming/OpenRefine/extensions/rdf-extension/MOD-INF/controller.js)
        at org.mozilla.javascript.gen.c10.exec(file:/C:/Users/tfmorris/AppData/Roaming/OpenRefine/extensions/rdf-extension/MOD-INF/controller.js)
        at edu.mit.simile.butterfly.ButterflyModuleImpl.initScope(ButterflyModuleImpl.java:619)
        at edu.mit.simile.butterfly.ButterflyModuleImpl.scriptInit(ButterflyModuleImpl.java:629)
        at edu.mit.simile.butterfly.ButterflyModuleImpl.init(ButterflyModuleImpl.java:94)
        at edu.mit.simile.butterfly.Butterfly.initializeModule(Butterfly.java:476)
        at edu.mit.simile.butterfly.Butterfly.configure(Butterfly.java:451)
        at edu.mit.simile.butterfly.Butterfly.init(Butterfly.java:308)
        at org.mortbay.jetty.servlet.ServletHolder.initServlet(ServletHolder.java:440)
        at org.mortbay.jetty.servlet.ServletHolder.doStart(ServletHolder.java:263)
        at com.google.refine.RefineServer.configure(Refine.java:291)
        at com.google.refine.RefineServer.init(Refine.java:203)
        at com.google.refine.Refine.init(Refine.java:109)
        at com.google.refine.Refine.main(Refine.java:103)
Caused by: java.lang.ClassNotFoundException: org.json.JSONException
        at java.net.URLClassLoader.findClass(Unknown Source)
        at java.lang.ClassLoader.loadClass(Unknown Source)
        at java.lang.ClassLoader.loadClass(Unknown Source)
        ... 34 more


### Expected behavior

Although it's generally discourage to catch Errors, in this case it makes sense to do so, although the exception handling may need to go in Butterfly rather than OpenRefine.

--- a/main/pom.xml
+++ b/main/pom.xml
@@ -173,7 +173,7 @@
     <dependency>
       <groupId>org.openrefine.dependencies</groupId>
       <artifactId>butterfly</artifactId>
-      <version>1.0.3</version>
+      <version>1.0.4</version>
     </dependency>
     <dependency>
       <groupId>org.mozilla</groupId>

--- a/server/pom.xml
+++ b/server/pom.xml
@@ -175,7 +175,7 @@
     <dependency>
       <groupId>org.openrefine.dependencies</groupId>
       <artifactId>butterfly</artifactId>
-      <version>1.0.3</version>
+      <version>1.0.4</version>
     </dependency>
     <dependency>
       <groupId>org.mozilla</groupId>

